File: src/main/java/com/github/codingvampyre/drachenfest/domain/BaseEntity.java
package com.github.codingvampyre.drachenfest.domain;

import lombok.Builder;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.Objects;

@MappedSuperclass
public abstract class BaseEntity implements Serializable {

    @GeneratedValue
    @Id
    private Long Id;

    @CreationTimestamp
    private Date createdAt;

    @UpdateTimestamp
    private Date updatedAt;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        BaseEntity that = (BaseEntity) o;

        return Objects.equals(Id, that.Id);
    }

    @Override
    public int hashCode() {
        return Id != null ? Id.hashCode() : 0;
    }
}

File: src/main/java/com/github/codingvampyre/drachenfest/domain/BaseEntity.kt
package com.github.codingvampyre.drachenfest.domain

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.io.Serializable
import java.util.*
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class BaseEntity : Serializable {
	@GeneratedValue
	@Id
	private val Id: Long? = null

	@CreationTimestamp
	private val createdAt: Date? = null

	@UpdateTimestamp
	private val updatedAt: Date? = null
	override fun equals(o: Any?): Boolean {
		if (this === o) return true
		if (o == null || javaClass != o.javaClass) return false
		val that = o as BaseEntity
		return Id == that.Id
	}

	override fun hashCode(): Int {
		return Id?.hashCode() ?: 0
	}
}